plotly/utils.py

```diff
@@ -399,3 +404,46 @@ def get_by_path(obj, path):
     for key in path:
         obj = obj[key]
     return obj
+
+
+### validation
+def validate_world_readable_and_sharing_settings(option_set):
+    if ('world_readable' in option_set and
+        option_set['world_readable'] is not None and
+        option_set['world_readable'] is True and
+        'sharing' in option_set and
+        option_set['sharing'] is not None and
+            option_set['sharing'] != 'public'):
+        raise PlotlyError(
+            "Looks like you are setting your plot privacy to both "
+            "public and private.\n If you set world_readable as True, "
+            "sharing can only be set to 'public'")
+    elif ('world_readable' in option_set and
+          option_set['world_readable'] is False and
+          'sharing' in option_set and
+          option_set['sharing'] == 'public'):
+        raise PlotlyError(
+            "Looks like you are setting your plot privacy to both "
+            "public and private.\n If you set world_readable as "
+            "False, sharing can only be set to 'private' or 'secret'")
+    elif ('sharing' in option_set and
+          option_set['sharing'] not in ['public', 'private', 'secret', None]):
+        raise PlotlyError(
+            "The 'sharing' argument only accepts one of the following "
+            "strings:\n'public' -- for public plots\n"
+            "'private' -- for private plots\n"
+            "'secret' -- for private plots that can be shared with a "
+            "secret url"
+        )
```

Comment:
At minimum, can we kill the redundant check on `is None` and `is True` or `is False`. You need at most one `is` check on a value in a conditional, right?